components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```diff
@@ -73,6 +90,115 @@ public Response getGroup(@PathParam(SCIMConstants.CommonSchemaConstants.ID) Stri
         return processRequest(requestAttributes);
     }
 
+    @GET
+    @Path("{id}/permissions")
+    @Produces({MediaType.APPLICATION_JSON, SCIMProviderConstants.APPLICATION_SCIM_JSON})
+    public Response getPermissionListOfGroup(@PathParam(SCIMConstants.CommonSchemaConstants.ID) String id,
+                                             @HeaderParam(SCIMProviderConstants.AUTHORIZATION) String authorizationHeader,
+                                             @HeaderParam(SCIMProviderConstants.CONTENT_TYPE) String inputFormat,
+                                             @HeaderParam(SCIMProviderConstants.ACCEPT_HEADER) String outputFormat) {
+
+        String userName = SupportUtils.getAuthenticatedUsername();
+        try {
+            if (!isValidOutputFormat(outputFormat)) {
+                throw new FormatNotSupportedException("Output format: " + outputFormat + " is not supported.");
+            }
+        } catch (FormatNotSupportedException e) {
+            return handleFormatNotSupportedException(e);
+        }
+
+        Map<String, String> requestAttributes = new HashMap<>();
+        requestAttributes.put(SCIMProviderConstants.ID, id);
+        requestAttributes.put(SCIMProviderConstants.ACCEPT_HEADER, outputFormat);
+        requestAttributes.put(SCIMProviderConstants.AUTHORIZATION, userName);
+        requestAttributes.put(SCIMProviderConstants.HTTP_VERB, GET.class.getSimpleName());
+        requestAttributes.put(SCIMProviderConstants.ATTRIBUTES, PERMISSIONS);
+        requestAttributes.put(SCIMProviderConstants.SEARCH, "0");
+        return processRequest(requestAttributes);
+    }
+
+    @PUT
+    @Path("{id}/permissions")
+    @Produces({MediaType.APPLICATION_JSON, SCIMProviderConstants.APPLICATION_SCIM_JSON})
+    public Response setPermissionForGroup(@PathParam(SCIMConstants.CommonSchemaConstants.ID) String id,
+                                          @HeaderParam(SCIMProviderConstants.AUTHORIZATION) String authorizationHeader,
+                                          @HeaderParam(SCIMProviderConstants.CONTENT_TYPE) String inputFormat,
+                                          @HeaderParam(SCIMProviderConstants.ACCEPT_HEADER) String outputFormat,
+                                          @QueryParam(SCIMProviderConstants.ATTRIBUTES) String attribute,
+                                          @QueryParam(SCIMProviderConstants.EXCLUDE_ATTRIBUTES) String excludedAttributes,
+                                          String resourceString) {
+
+        String userName = SupportUtils.getAuthenticatedUsername();
+        try {
```

Comment:
Throwing and catching in the same try-catch is a good practice. handleFormatNotSupportedException directly instead of throwing and catching.

if (!isValidOutputFormat(outputFormat)) {
return handleFormatNotSupportedException(e);
}